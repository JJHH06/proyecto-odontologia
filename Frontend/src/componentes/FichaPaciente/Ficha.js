
import React, { useState } from 'react';
import { useLocation } from "react-router-dom";
import './Ficha.scss';
import Odontogram from "../Odontograma/Odontogram"


function Ficha(props) {

    const location = useLocation();
    console.log(location.state);
    //console.log(props.location.state)
    return (
        <div className= 'container estilo-ficha'>
            <h1>{location.state.nombre}</h1>
            <div className='row'>
                <div className='col-lg-6'>
                    <div className='cuadro'>
                        <div className='container-h3'>
                            <h3>Paciente</h3>
                        </div>

                        <ul>
                            <li><p className='subtitulo'>Estado Civil: <span>{location.state.estado_civil}</span></p> </li>
                            <li><p className='subtitulo'>Edad: <span>{location.state.edad}</span></p> </li>
                            <li><p className='subtitulo'>Fecha de Nacimiento: <span>{location.state.fecha_nacimiento}</span></p> </li>
                            <li><p className='subtitulo'>Ocupacion: <span>{location.state.ocupacion}</span></p> </li>
                        </ul>
                    </div>
                </div>
                <div className='col-lg-6'>
                    <div className='cuadro'>
                        <div className='container-h3'>
                            <h3>Contacto</h3>
                        </div>
                        <ul>
                            <li><p className='subtitulo'>Direccion: <span>{location.state.direccion}</span></p> </li>
                            <li><p className='subtitulo'>Email: <span>{location.state.email}</span></p> </li>
                            <li><p className='subtitulo'>Telefono Celular: <span>{location.state.telefono_celular}</span></p> </li>
                            <li><p className='subtitulo'>Telefono de Casa: <span>{location.state.telefono_casa}</span></p> </li>
                        </ul>
                    </div>
                </div>
                <div className='col-lg-12'>
                    <div className='cuadro'>
                        <div className='container-h3 grande'>
                            <h3>Historial Medico</h3>
                        </div>
                        <ul>
                            <li><p className='subtitulo'>Recomendado Por: <span>{location.state.recomendado_por}</span></p> </li>
                            <li><p className='subtitulo'>Visita anterior al dentista: <span>{location.state.visita_anterior_dentista}</span></p> </li>
                            <li><p className='subtitulo'>Medico: <span>{location.state.medico}</span></p> </li>
                            <li><p className='subtitulo'>Telefono medico: <span>{location.state.telefono_medico}</span></p> </li>
                            <li><p className='subtitulo'>Contacto de emergencia: <span>{location.state.contacto_emergencia}</span></p> </li>
                            <li><p className='subtitulo'>Telefono medico: <span>{location.state.telefono_medico}</span></p> </li>
                            
                            
                        </ul>
                    </div>
                </div>
                <div className='col-lg-12'>
                    <div className='cuadro'>
                        <div className='container-h3'>
                            <h3>Cita</h3>
                        </div>
                        <ul>
                            <li><p className='subtitulo'>Motivo de consulta: <span>{location.state.motivo_consulta}</span></p> </li>
                            <li><p className='subtitulo'>Presupuesto: <span>No calculado</span></p> </li>
                            <li><p className='subtitulo'>Medicamentos: <span>{location.state.medicamentos}</span></p> </li>
                            <li><h3>Condiciones preexistentes</h3></li>
                            <li><ul>
                            {
                                location.state.padecimientos.map((padecimiento,index) =>(
                                    <li><p>{padecimiento}</p></li>
                                ))
                            }
                            </ul></li>
                            
                        </ul>
                    </div>
                    <div className='cuadro'>
                        <Odontogram id_paciente={location.state.id_paciente}/>
                    </div>
                </div>
                
            </div>
        </div>
    );
}

export default Ficha;
