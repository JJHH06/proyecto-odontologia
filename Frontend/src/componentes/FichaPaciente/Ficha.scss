.container.estilo-ficha{
    margin-top: 20px;
}

.container.estilo-ficha h1{
    text-align: center;
}



.container.estilo-ficha .container-h3{
    align-items: center;
    justify-content: center;
    text-align: center;
    border: solid 1px;
    width: 150px;
    background-color: rgb(168, 255, 255);
    border-radius: 5px 5px 5px 5px;
    transform:translateY(-20px);
}

.container.estilo-ficha .col-lg-6 .cuadro{
    border: solid 2px;
    margin-top: 50px;
    box-shadow: 1px 1px 1px 1px rgba(0, 0, 0, 0.5);
    padding-left: 20px;
    height: 300px;

}

.container.estilo-ficha .subtitulo{
    font-weight: bold;
}

.container.estilo-ficha span{
    font-weight: 400;
}

.container.estilo-ficha li{
    list-style: none;
}