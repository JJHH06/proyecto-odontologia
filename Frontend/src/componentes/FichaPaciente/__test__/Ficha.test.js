import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {render, cleanup, screen} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Ficha from '../Ficha';
import { jssPreset } from '@material-ui/styles';


jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
    location: {state:"Elpepe"},
      pathname: '/another-route',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
    }),
}));

afterEach(cleanup);
it("Renders without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Ficha/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("Renders Ficha Correctly",() =>{
    const {getByTestId}= render(<Ficha/>)
    expect(getByTestId('patient-form')).toHaveTextContent("Paciente")
});

it('Ficha result is displayed', async () => {
    render(<Ficha/>);
    expect(screen.getByTestId('patient-form')).toBeInTheDocument();
});

it("Matches snapshot",() =>{
    const tree = renderer.create(<Ficha/>).toJSON();
    expect(tree).toMatchSnapshot();
})
