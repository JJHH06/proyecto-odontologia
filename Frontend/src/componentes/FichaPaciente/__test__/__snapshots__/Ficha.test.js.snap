// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  data-testid="patient-form"
>
  Paciente no disponible, La vuelve a la pantalla de inicio
</div>
`;
